src/nvim/charset.c

```diff
@@ -399,33 +400,22 @@ char_u *transstr(char_u *s) FUNC_ATTR_NONNULL_RET
 char_u* str_foldcase(char_u *str, int orglen, char_u *buf, int buflen)
   FUNC_ATTR_NONNULL_RET
 {
-  garray_T ga;
+  kstring_t ks = KSTRING_INIT;
   int i;
   int len = orglen;
 
-#define GA_CHAR(i) ((char_u *)ga.ga_data)[i]
-#define GA_PTR(i) ((char_u *)ga.ga_data + i)
-#define STR_CHAR(i) (buf == NULL ? GA_CHAR(i) : buf[i])
-#define STR_PTR(i) (buf == NULL ? GA_PTR(i) : buf + i)
+#define STR_CHAR(i) (buf == NULL ? (char_u)ks.s[i] : buf[i])
+#define STR_PTR(i) (buf == NULL ? (char_u *)ks.s + i : buf + i)
 
-  // Copy "str" into "buf" or allocated memory, unmodified.
+  // Copy "str" into "buf" or allocated memory(kstring), unmodified.
   if (buf == NULL) {
-    ga_init(&ga, 1, 10);
-
-    ga_grow(&ga, len + 1);
-    memmove(ga.ga_data, str, (size_t)len);
-    ga.ga_len = len;
+    kputsn((char *)str, len, &ks);
```

Comment:
If we converted garray to a wrapper around kstring, the `memmove` above would be unwanted. But it would be much easier to search the code for `ga_grow` calls and clean them up (there's only 71), than to clean up all garray-related code. If that proves to be a performance boost then there will be more enthusiasm for removing garray entirely.